src/scm/github.js

```diff
@@ -0,0 +1,427 @@
+        context.repo = repo;
+      } else if (context.gist) {
+        context.repo = {
+          name: 'Using Gist',
+          fullName: 'gist',
+          gist: true
+        }
+      }
+      return repos;
+    })
+  }
+
+  createRepo() {
+    const repo = $('#new-repo-name').val();
+    const desc = $('#new-repo-desc').val();
+    const isPrivate = $('#new-repo-type').val() !== 'public';
+    const payload = {
+      name : repo,
+      description : desc,
+      auto_init : true,
+      private: isPrivate
+    }
+    if (!repo || repo === '') return;
+    new Promise((resolve, reject) => {
+      $.ajax({
+        url: `${baseUrl}/user/repos`,
+        headers: {
+          'Authorization': `token ${accessToken}`
+        },
+        method: 'POST',
+        crossDomain: true,
+        dataType: 'json',
+        contentType: 'application/json',
+        data: JSON.stringify(payload)
+      })
+      .then(resolve)
+      .fail(reject);
+    })
+    .then((response) => {
+      const repo = {
+        name : response.name,
+        fullName : response.full_name
+      };
+      context.repo = repo;
+      Object.assign(context.bindRepo, { [context.id] : repo });
+      if (context.bindBranch[context.id]) {
+        delete context.bindBranch[context.id];
+      }
+      chrome.storage.sync.set({ bindRepo: context.bindRepo });
+      return response;
+    })
+    .then(getGithubRepos)
+    .then(updateRepo)
+    .then(updateBranch)
+    .then(() => {
+      $('#new-repo-name').val('');
+      $('#new-repo-desc').val('');
+      $('#new-repo-type').val('public');
+      showAlert(`Successfully create new repository ${repo}`);
+    })
+    .catch((err) => {
+      showAlert('Failed to create new repository.', LEVEL_ERROR);
+    });
+  }
+
+  createGist() {
+    const desc = $('#new-gist-name').val();
+    const isPublic = $('#new-gist-public').val() !== 'secret';
+    const payload = {
+      'description' : desc,
+      'public': isPublic,
+      'files': {
+        'init_by_gas_hub.html' : {
+          'content': 'init by gas-hub, just delete this file.'
+        }
+      }
+    };
+
+    new Promise((resolve, reject) => {
+      $.ajax({
+        url: `${baseUrl}/gists`,
+        headers: {
+          'Authorization': `token ${accessToken}`
+        },
+        method: 'POST',
+        crossDomain: true,
+        dataType: 'json',
+        contentType: 'application/json',
+        data: JSON.stringify(payload)
+      })
+      .then(resolve)
+      .fail(reject);
+    })
+    .then((response) => {
+      const gist = response.id;
+      context.branch = gist;
+      Object.assign(context.bindBranch, { [context.id] : gist });
+      chrome.storage.sync.set({ bindBranch: context.bindBranch });
+      return response;
+    })
+    .then(updateGist)
+    .then(() => {
+      $('#new-gist-name').val('');
+      $('#new-gist-public').val('public');
+      showAlert(`Successfully create new gist.`);
+    })
+    .catch((err) => {
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/leonhartX/gas-github/blob/9b7bf2b597e67365f1e9002c784237b58657e4cc/src/scm/github.js#L361"></span>‘err’ is defined but never used.</td>
    </tr>
  </tbody>
</table>